lib/eth_web_socket/client.ex

```diff
@@ -0,0 +1,19 @@
+defmodule EthWebSocket.Client do
+  use WebSockex
+  require Logger
+
+  def start_link(state, opts \\ []) do
+    # Find a better way to get env var
+    WebSockex.start_link(System.get_env("ETH_WEB_SOCKET"), __MODULE__, state, opts)
+  end
+
+  def handle_frame({:text, msg}, state) do
+    send(state.gs_pid, {:asynchronous_request_response, msg})
```

Comment:
How do you know the `state` has `gs_pid`?

Reply:
I know that the state has gs_pid because, when I start the websocket, I pass the GenServer process ID as an argument to the initial state.

`{:ok, ws_pid} = Client.start_link(%{gs_pid: gs_pid})`

Reply:
So, I need to know all the code to understand what values `state` holds. 
Couldn't this be improved?
Isn't there a way to say that the `state` will be something fixed?

Reply:
I think that it is better now

Reply:
When I said something fixed, I suggested creating a fixed state struct for the GenServer state. This way, I can go to the State whenever I'm lost and see precisely what it is supposed to have.

Reply:
I believe it's improved now; in the latest commit, I utilized a struct to define both the GenServer and the WebSocket state 😁